jupyterlab_automl/jupyterlab_automl/service.py

```diff
@@ -321,4 +366,165 @@ def export_dataset(self, dataset_id):
+      if not os.path.isfile(local_file):
+        continue
+      remote_path = os.path.join(gcs_path, local_file[1 + len(local_path):])
+      blob = bucket.blob(remote_path)
+      blob.upload_from_filename(local_file)
+
+  def import_saved_model(self,
+                         display_name,
+                         model_path,
+                         framework):
+    # Strip out non alphanumeric chars of model name and prepend uuid
+    name = "{}-{}".format(re.sub(r'\W+', '', display_name), str(uuid.uuid4()))
+    key = "models/" + name
+    bucket = self._get_gcs_bucket()
+
+    # Upload model artifacts to GCS
+    self._copy_local_directory_to_gcs(model_path, bucket, key)
+
+    # https://pantheon.corp.google.com/gcr/images/cloud-aiplatform/GLOBAL/prediction?gcrImageListsize=30
+    if not isinstance(framework, str):
+      framework = framework.value
+
+    container_spec = aiplatform_v1alpha1.ModelContainerSpec(
+        image_uri="gcr.io/cloud-aiplatform/prediction/{}:latest".format(
+            framework))
+    model = aiplatform_v1alpha1.Model(display_name=display_name,
+                                      artifact_uri="gs://{}/{}".format(
+                                          bucket.name, key),
+                                      container_spec=container_spec)
+    return self._model_client.upload_model(parent=self._parent,
+                                           model=model)
+
+  def get_dataset_details(self, dataset_id):
+    df = self.export_dataset(dataset_id)
+    columns = list(df.columns)
+    return [{
+      "fieldName": column
+    } for column in columns]
+
+  def create_training_pipeline(
+      self,
+      training_pipeline_name,
+      dataset_id,
+      model_name,
+      target_column,
+      prediction_type,
+      objective,
+      budget_hours,
+      transformations,
+  ):
+    training_task_inputs = {
+        "targetColumn": target_column,
+        "predictionType": prediction_type,
+        "transformations": transformations,
+        "trainBudgetMilliNodeHours": budget_hours * 1000,
+        "disableEarlyStopping": False,
+        "optimizationObjective": objective,
+    }
+    training_pipeline = {
+        "display_name": training_pipeline_name,
+        "training_task_definition": "gs://google-cloud-aiplatform/schema/trainingjob/definition/automl_tables_1.0.0.yaml",
+        "training_task_inputs": json_format.ParseDict(training_task_inputs, Value()),
+        "input_data_config": {
+            "dataset_id": dataset_id,
+            "fraction_split": {
+                "training_fraction": 0.8,
+                "validation_fraction": 0.1,
+                "test_fraction": 0.1,
+            },
+        },
+        "model_to_upload": {"display_name": model_name},
+    }
+    response = self._pipeline_client.create_training_pipeline(
+        parent=self._parent, training_pipeline=training_pipeline
+    )
+    print(" training_display_name:", response.display_name)
```

Comment:
It's not really for debugging but to give the user feedback on the creation of the training pipeline. Though on second thought this could all probably just be returned from the function instead of printed.